### ISSUE TITLE
Add view ports in the skeleton-navigation 

### ISSUE THREAD

**Issue by EisenbergEffect** [aurelia/router#339]
@ceoaliongroo commented on [Tue May 26 2015](https://github.com/aurelia/skeleton-navigation/issues/101)

I'm using view ports to refactoring a ui-router application.

I did an example, to check. How to use the view ports in the skeleton-navigation  code. 
https://github.com/ceoaliongroo/sandbox-aurelia/pull/6/files

``` javascript
export class App {
  configureRouter(config, router){
    config.title = 'Dashboard';
    config.map([
      {
        route: ['','dashboard'],
        viewPorts: {
          left: {
            moduleId: 'left'
          },
          right: {
            moduleId: 'dashboard/right'
          }
        },
        nav: true,
        title:'Dashboard'
      },
      {
        route: ['more'],
        viewPorts: {
          left: {
            moduleId: 'left'
          },
          right: {
            moduleId: 'dashboard/other'
          }
        },
        nav: true,
        title:'Dashboard with more content'
      }
    ]);

    this.router = router;
  }
}
```

This code works fine, except the navigation menu, each moduleId id take as a router it self.  Go to the repository/branch: https://github.com/ceoaliongroo/sandbox-aurelia/tree/5

Is this the behaviour as expected? or there is a possibility to create a route to ech view port?

![screen shot 2015-05-26 at 10 05 54 am](https://cloud.githubusercontent.com/assets/2622152/7807082/01538532-0391-11e5-9499-8ca069ff58fb.png)
![screen shot 2015-05-26 at 10 04 51 am](https://cloud.githubusercontent.com/assets/2622152/7807083/015489be-0391-11e5-8999-33cf1056c150.png)

---

@EisenbergEffect commented on [Mon Jun 01 2015](https://github.com/aurelia/skeleton-navigation/issues/101#issuecomment-107774500)

I'm not sure I understand the question. Can you rephrase it please?

---

@ceoaliongroo commented on [Mon Jun 01 2015](https://github.com/aurelia/skeleton-navigation/issues/101#issuecomment-107778044)

Ok, 
1. \- Why the `navigation component` show an item for a each `viewport` element instead, each `route`?
   https://github.com/ceoaliongroo/sandbox-aurelia/blob/master/src/nav-bar.html#L18-L20

**route definition:**

``` javascript
{
        route: ['','dashboard'],
        viewPorts: {
          left: {
            moduleId: 'left'
          },
          right: {
            moduleId: 'dashboard/right'
          }
        },
        nav: true,
        title:'Dashboard'
      },
      {
        route: ['more'],
        viewPorts: {
          left: {
            moduleId: 'left'
          },
          right: {
            moduleId: 'dashboard/other'
          }
        },
        nav: true,
        title:'Dashboard with more content'
      }
```

**Another question related:**

1.- It's possible to define the viewport  `moduleId: left` once, for all the routes?
2.- It's possible to define a subroute to change only change the view port right, with out create a new route?. 

Thanks

---

@EisenbergEffect commented on [Mon Jun 01 2015](https://github.com/aurelia/skeleton-navigation/issues/101#issuecomment-107778969)

@bryanrsmith See above. It may be that we have a bug with the navigation model when we've got view ports defined. Not the highest priority, but when you get other things off your plate, let's look into this.

@ceoaliongroo It's not possible to define the left value once for all the routes. If you want to do that, you might consider creating your own helper function to create the route object with that left value set automatically for each. Regarding subroutes, you might want to look into using child routers.

---

@PWKad commented on [Sun Jan 10 2016](https://github.com/aurelia/skeleton-navigation/issues/101#issuecomment-170365388)

Should we move this to the router so we can triage appropriately?

---

@EisenbergEffect commented on [Sun Jan 10 2016](https://github.com/aurelia/skeleton-navigation/issues/101#issuecomment-170365763)

Yes, probably ;)

**Comment by gregorydickson** [comment]
+1 this, I am also in the dashboard model where I don't want to use routes but have a dashboard with several areas/widgets/viewports.

**Comment by EisenbergEffect** [comment]
If you don't want to use routes, then don't use the router. Use the compose element instead.

**Comment by gregorydickson** [comment]
ah, thanks!!

**Comment by Alexander-Taran** [comment]
can be closed
